lib/assets/javascripts/src/local-time/helpers/strftime.coffee

```diff
@@ -40,6 +50,23 @@ pad = (num, flag) ->
     else ("0#{num}").slice(-2)
 
 parseTimeZone = (time) ->
+  if longTimeZone = edgeCaseTimeZone(time)
+    knownEdgeCaseTimeZones[longTimeZone]
+  else if shortTimeZone = parseTimeZoneWithIntl(time)
+    shortTimeZone
+  else
+    parseTimeZoneWithHeuristic(time)
+
+edgeCaseTimeZone = (time) ->
+  Object.keys(knownEdgeCaseTimeZones).find (name) ->
+    time.toString().indexOf(name) isnt -1
```

Comment:
We have to match on the long time zone name because there are some geographies which share the same abbreviation for different time zones. From ChatGPT (which wouldn't give me an exhaustive list):

* PST:
  * Pacific Standard Time (UTC-8): observed in parts of the United States and Canada.
  * Philippine Time (UTC+8): observed in the Philippines.
* EST:
  * Eastern Standard Time (UTC-5): observed in parts of the United States and Canada.
  * Eastern Standard Time (UTC+10): observed in parts of Australia during their standard time.
  * _Note: glad we don't have to match on this one since they even have the same long name_
* MST:
  * Mountain Standard Time (UTC-7): observed in parts of the United States and Canada.
  * Malaysia Standard Time (UTC+8): observed in Malaysia.
* IST:
  * Indian Standard Time (UTC+5:30): observed in India.
  * Irish Standard Time (UTC+1): observed in Ireland during standard time (UTC+0 during daylight saving time).
* CAT:
  * Central Africa Time (UTC+2): observed in several African countries.
  * Central Alaska Time (UTC-9): observed in some areas of Alaska, USA.
* AST:
  * Atlantic Standard Time (UTC-4): observed in parts of Canada and the Caribbean.
  * Arabia Standard Time (UTC+3): observed in the Arabian Peninsula.
* GMT:
  * Greenwich Mean Time (UTC+0): observed in the United Kingdom and other regions.
  * Gambia Mean Time (UTC+0): observed in The Gambia.
* HST:
  * Hawaii Standard Time (UTC-10): observed in Hawaii, USA.
  * Honduras Standard Time (UTC-6): observed in Honduras.
* CST:
  * Central Standard Time (UTC-6): observed in parts of North America, including parts of the United States and Canada.
  * China Standard Time (UTC+8): observed in China and other regions.